src/hyrdate.js

```diff
@@ -1,20 +1,30 @@
 class Hydration {
-  constructor(hydrationData) {
-    this.userId = hydrationData.userID,
-    this.date = hydrationData.date,
-    this.numOunces = hydrationData.numOunces
+  constructor(hydrationData, userID) {
+    this.hydrationData = hydrationData;
+    this.userID = userID;
   }
 
-  calculateAverageDailyConsumption() {
+  calculateTotalConsumption(id) {
+    let totalConsumption = this.hydrationData.reduce((acc, total) => {
+      if(total.userID === id) {
+        acc += total.numOunces;
+      }
+      return acc
+    }, 0)
+    return totalConsumption;
+      }
 
-  }
 
-  returnDailyConsumption() {
 
+  returnDailyConsumption(date) {
+  let dailyConsumption = this.hydrationData.find(consumption => {
+       return consumption.date === date;
+    })
+     return dailyConsumption.numOunces;
   }
 
   returnWeeklyConsumption() {
-
+    return this.hydrationData.slice(-7).map(day => day.numOunces);
```

Comment:
This one will also need two parameters/arguments, one being the particular week that is needing to be returned and the other being the user ID.  In your testing data set, you only have three days worth of info included in the dataset but we are likely going to need a lot more than that.  I think ten would be about the minimum amount since it will require 7 days worth of info for one particular ID and additional days worth of info spaced in between that but from a different user.  This is to prove in the testing that it's actually doing what it's supposed to do.  You will also need to include a week variable to pass as the argument in this function.

Reply:
I have corrected the issues with hydration. I reread the requirements for weekly consumption and It just says calculate for 7 days. It does not specify that it needs to be able to take in a particular date. I have it so that it will display 7 days worth of oz.